### ISSUE TITLE
ORA-03149 and ORA-13149

### ISSUE THREAD

**Issue by andreynering** [mattn/go-oci8#119]
I'm getting these two errors on some queries. They are very generic, I didn't found any useful information on the web, about the cause or a possible solution.

This might not be any bug with this driver, but I'm posting to know if someone has any cue. 😞

**Comment by mattn** [comment]
What did you do? code? your os? your environment? go version?

**Comment by andreynering** [comment]
Sorry for the missing information

> What did you do? code?

It's a simple select statement. It's big, but have no errors because runs on my SQL editor. Unfortunally I can't post here. Other smaller SQLs run without error.

> your os? your environment? go version?

- Windows Server 2012 x64
- Latest Go x86
- Oracle Client 11g x86
- Server also 11g

I think it have something to do with the server configuration. I don't get this error on my local Oracle setup which uses the default options. But in this customer, Oracle was installed with non-default options, like a different collation: `WE8ISO8859P1` (I'll ask him for any other different options). For other languages this error doesn't happen.

**Comment by andreynering** [comment]
![1](https://cloud.githubusercontent.com/assets/7011819/20386529/32565c6a-aca3-11e6-87f9-d3123ac896b7.png)

Do you think any different config can fix this error? Maybe we should force any config as we forced collation (#118)?

Unfortunally, I was still unable to reproduce this locally.

**Comment by andreynering** [comment]
I could workaround the issue.

I had to get rid of the parameters and concatenate the SQL myself. Maybe the driver is getting lost on types (e.g. in Go it's `int` but in server is `smallint` or `number` and not `integer`).

After that I got this error:

```
Scan error on column index 0: converting driver.Value type string("1102,82") to a float64: invalid syntax
```

But that was fixed by forcing `.` on floats with:

```go
os.Setenv("NLS_NUMERIC_CHARACTERS", ".,")
```

After these two changes everything is finally working.
